File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    s_li = list(s)
    li = []
    while s_li:
        new_s1 = s_li.pop()
        if len(li) == 0:
            li.append(new_s1)
        else:
            new_s2 = li.pop()
            if new_s1 == "(" and new_s2 == ")":
                continue
            else:
                li.append(new_s2)
                li.append(new_s1)
    if len(li) != 0:
        return False
    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    var answer = true;
    
    const s_li = s.split('')
    const li = Array()
    
    while (s_li.length > 0) {
        const new_s1 = s_li.pop()
        if (li.length === 0) {
            li.push(new_s1)
        } else {
            const new_s2 = li.pop()
            if (new_s1 === "(" && new_s2 === ")") {
                continue
            } else {
                li.push(new_s2)
                li.push(new_s1)
            }
        }
    }
    if (li.length !== 0) {
        return false
    }
    return answer;
}